Development/nmos/api_downgrade.cpp

```diff
@@ -1,11 +1,973 @@
 #include "nmos/is04_versions.h"
 #include "nmos/resource.h"
+#include "nmos/type.h"
 
 namespace nmos
 {
+    namespace details
+    {
+        static web::json::value make_schema(const char* schema)
+        {
+            return web::json::value::parse(utility::s2us(schema));
+        }
+
+        static web::uri make_schema_id(const api_version& version, const nmos::type& type, const utility::string_t& property)
+        {
+            return U("/downgrade") + make_api_version(version) + U("-") + type.name + U("-") + property;
+        }
+
+        // property schemas are extracted from the resource schemas
+        const std::map<web::uri, web::json::value> property_schemas =
+        {
+            // node
+            // services
+            { make_schema_id(is04_versions::v1_0, nmos::types::node, U("services")), make_schema(R"({
+                "$schema": "http://json-schema.org/draft-04/schema#",
+                "type": "object",
+                "properties": {
+                  "services": {
+                    "description": "Array of objects containing a URN format type and href",
+                    "type": "array",
+                    "items": {
+                      "type": "object",
+                      "required": ["href", "type"],
+                      "properties": {
+                        "href": {
+                          "type": "string",
+                          "description": "URL to reach a service running on the Node",
+                          "format": "uri"
+                        },
+                        "type": {
+                          "type": "string",
+                          "description": "URN identifying the type of service",
+                          "format": "uri"
+                        }
+                      }
+                    }
+                  }
+                }
+              })")
+            },
+            { make_schema_id(is04_versions::v1_3, nmos::types::node, U("services")), make_schema(R"({
+                "$schema": "http://json-schema.org/draft-04/schema#",
+                "type": "object",
+                "properties": {
+                  "services": {
+                    "description": "Array of objects containing a URN format type and href",
+                    "type": "array",
+                    "items": {
+                      "type": "object",
+                      "required": ["href", "type"],
+                      "properties": {
+                        "href": {
+                          "type": "string",
+                          "description": "URL to reach a service running on the Node",
+                          "format": "uri"
+                        },
+                        "type": {
+                          "type": "string",
+                          "description": "URN identifying the type of service",
+                          "format": "uri"
+                        },
+                        "authorization": {
```

Comment:
Is it necessary to remove `authorization` when downgrading before v1.3? The v1.2 schema doesn't consider it invalid as far as I know. Same applies to many other cases.